hummingbot/client/settings.py

```diff
@@ -340,4 +340,4 @@ def gateway_connector_trading_pairs(connector: str) -> List[str]:
 MAXIMUM_TRADE_FILLS_DISPLAY_OUTPUT = 100
 
 STRATEGIES: List[str] = get_strategy_list()
-GATEWAY_CONNECTORS: List[str] = ["pangolin", "uniswap"]
+GATEWAY_CONNECTORS: List[str] = []
```

Comment:
`GATEWAY_CONNECTORS` is used in `completer.py` to initialize `_gateway_connect_completer` which is used for `gateway connect` autocompletions. Removing it will affect the autocomplete suggestions.

Reply:
of course.
That's why it's repopulated whenever there's a connection to gateway.

So here's the point, a user might switch between different of gateway in the future. Hardcoding a list of gateway connectors  on the client isn't flexible enough because different version can have different list of implemented connectors. Hence, I've redesigned this to be more flexible and repopulate that list of connectors every time a connection is re-established to gateway.

So you may want to argue that `gateway connect` wouldn't have any completion if no connection has been established with gateway at first. But the fact is that `gateway connect ...` wouldn't work anyway because gateway is yet to be started.

Let me know if you still have questions or doubt. Thanks.